### DISCUSSION TITLE
Trouble excluding maven-surefire-plugin

### DISCUSSION THREAD

**Discussion by LakerBaker** [renovatebot/renovate#19704] [Request Help] [answered]
I have a rather simple renovate.json5 file that includes the following snippet:

 "packageRules": [
    {
      "matchPackagePatterns": [
        "*"
      ],
      "excludePackageNames": [
        "maven-surefire-plugin"
      ],
      "matchUpdateTypes": [
        "patch",
        "minor"
      ],
      "groupName": "everything weekly",
      "schedule": [
        "every monday",
      ]
    },
  ],

My desire is to NOT update maven-surefire-plugin, but I have not been successful in doing so. Any tips?

And can you tell me why renovatebot doesn't complain about the &ast; value of matchPackagePatterns? It's a glob expression, not a regular expression. (I've tried to solve my issue by using excludePackagePatterns, and found when doing so that renovatebot complains about a value of &ast;maven-surefire-plugin&ast; but does not complain about a value of .&ast;maven-surefire-plugin.&ast;). 

A snippet of pom.xml might be useful:

<build>
        <plugins>
                <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>

**Comment by szpak** [comment, marked as answer]
From the documentation for [package rules](https://docs.renovatebot.com/configuration-options/#packagerules):

> For Maven dependencies, the package name is <groupId:artefactId>, e.g. "matchPackageNames": ["com.thoughtworks.xstream:xstream"]

Have you tried with the `groupId:articactId` in `matchPackageNames`?

**Reply by LakerBaker** [reply]
Yes, that did it. Marking as answer. Thanks.

**Comment by viceice** [comment]
please format your config as JSON
